Framework/Algorithms/test/ConvertDiffCalTest.h

```diff
@@ -33,6 +36,190 @@ class ConvertDiffCalTest : public CxxTest::TestSuite {
     TS_ASSERT(alg.isInitialized());
   }
 
+  /* specify fake entry in offset workspace or calibration table workspace */
+  class fake_entry
+  {
+    public:
+
+    enum
+    {
+      offset, /* fake entry specifies an entry in the fake input offsets workspace */
+      masked, /* fake entry specifies masked offset workspace entry */
+      unmasked,
+      calibration, /* fake entry specifies an entry in the input calibration table */
+    };
+
+    fake_entry( int detector_id,
+                int workspace_type,
+                double difc,
+                int mask = fake_entry::unmasked,
+                double difa = 0,
+                double tzero = 0 )
+      :
+      detector_id(detector_id),
+      workspace_type(workspace_type),
+      difc(difc),
+      mask(mask),
+      difa(difa),
+      tzero(tzero)
+    {}
+
+    int detector_id;
+    int workspace_type;
+    double difc;
+    int mask;
+    double difa;
+    double tzero;
+  };
+
+  /* contains a fake offset workspace and accompanying fake previous calibration workspace */
+  class fake_workspaces
+  {
+    public:
+
+    fake_workspaces( OffsetsWorkspace_sptr offsets, ITableWorkspace_sptr calibration_table )
+      :
+      offsets( offsets ),
+      calibration_table( calibration_table )
+    {}
+
+    OffsetsWorkspace_sptr offsets;
+
+    ITableWorkspace_sptr calibration_table; 
+  };
+
+  fake_workspaces generate_test_data( std::list< class fake_entry > const &entries )
+  {
+    Mantid::Geometry::Instrument_sptr instrument =
+    std::make_shared<Mantid::Geometry::Instrument>();
+
+    ITableWorkspace_sptr calibration_table =
+    std::make_shared<Mantid::DataObjects::TableWorkspace>(); 
+    calibration_table->addColumn("int", "detid");
+    calibration_table->addColumn("double", "difc");
+    calibration_table->addColumn("double", "difa");
+    calibration_table->addColumn("double", "tzero");
+
+    /* loop to add detectors and ids to instrument */
+    for( auto const &entry : entries )
+    {
+      /* add an entry to the fake offset workspace by adding a detector there */
+      if( entry.workspace_type == fake_entry::offset )
+      {
+        /* create a detector. Is this a memory leak? Idk. Depends what instrument does with it */
+        Mantid::Geometry::Detector *det =
+        new Mantid::Geometry::Detector("point-detector", entry.detector_id, nullptr);
+        instrument->add(det);
+        instrument->markAsDetector(det);
+      }
+
+      /* add an entry to the fake calibration workspace */
+      else if( entry.workspace_type == fake_entry::calibration )
+      {
+        Mantid::API::TableRow new_row = calibration_table->appendRow();
+        new_row << entry.detector_id << entry.difc << entry.difa << entry.tzero;
+      }
+    }
+
+    /* create an offset workspace with the instrument */
+    OffsetsWorkspace_sptr offsets = std::make_shared<OffsetsWorkspace>(instrument);
+    Mantid::Geometry::DetectorInfo &d_info = offsets->mutableDetectorInfo();
```

Comment:
This all looks fine so far except that I think the masking of the OffsetsWorkspace is done by spectrum and not detector. So you should be using `spectrumInfo` instead of `detectorInfo` to set and check the mask.

Reply:
done!